File: experimental/qldb-specification/tuple-hash.go
package main

import (
	"crypto/md5"
	"encoding/hex"
	"encoding/json"
	"fmt"
)

func main() {
	atl_L := [][2]interface{}{
		{1, "s1"},
		{2, "str"},
	}

	var sl_hash []string

	for _, item := range atl_L {
		jsonBytes, err := json.Marshal(item)
		if err != nil {
			panic(err)
		}
		sum := md5.Sum(jsonBytes)
		sl_hash = append(sl_hash, hex.EncodeToString(sum[:]))
	}

	fmt.Println(sl_hash)
}

File: experimental/qldb-specification/tuple-hash.js
const crypto = require("crypto");

const atl_L = [[1, "s1"], [2, "str"]];
const sl_hash = atl_L.map(item => {
    const json = JSON.stringify(item);
    return crypto.createHash("md5").update(json).digest("hex");
});

console.log(sl_hash);
